answerkey/state/11.answer.scala

```diff
@@ -14,7 +14,10 @@ object Candy {
         Machine(false, candy, coin + 1)
       case (Turn, Machine(false, candy, coin)) =>
         Machine(true, candy - 1, coin)
-    })))
+    }
+
+  def simulateMachine(inputs: List[Input]): State[Machine, (Int, Int)] = for {
+    _ <- sequence(inputs map update map modify)
```

Comment:
Yes, you are right.
I wondered which code would be a good one because `modify` requires some tricks like `(modify[Machine] _ compose update)` to combine. But for sure, it's wasteful to use multiple `map` just like you pointed out. I will fix this.
Thanks!